compiler/rustc_codegen_ssa/src/back/symbol_export.rs

```diff
@@ -44,7 +44,7 @@ pub fn crates_export_threshold(crate_types: &[CrateType]) -> SymbolExportLevel {
 }
 
 fn reachable_non_generics_provider(tcx: TyCtxt<'_>, _: LocalCrate) -> DefIdMap<SymbolExportInfo> {
-    if !tcx.sess.opts.output_types.should_codegen() {
+    if !tcx.sess.opts.output_types.should_codegen() && !tcx.is_interface_build() {
```

Comment:
Do we even get to this logic in `--emit=metadata` builds?

Reply:
Yes, see `encode_exported_symbols`.

Reply:
I don't see it there, but apparently it's used through `encode_exported_symbols` or something like that.